README.md
SYNO.phpMyBackupPro
===================

Synology package to install phpMyBackupPro.

Also available at community package hub repository http://www.cphub.net/


src/CHANGELOG
Version 2.3-001 - 10/03/2013
- Initial package release.

src/phpMyBackupPro/javascripts.js
<script type="text/javascript">
/*
 +--------------------------------------------------------------------------+
 | phpMyBackupPro                                                           |
 +--------------------------------------------------------------------------+
 | Copyright (c) 2004-2012 by Dirk Randhahn                                 |                               
 | http://www.phpMyBackupPro.net                                            |
 | Version information can be found in definitions.php.                     |
 |                                                                          |
 | This program is free software; you can redistribute it and/or            |
 | modify it under the terms of the GNU General Public License              |
 | as published by the Free Software Foundation; either version 2           |
 | of the License, or (at your option) any later version.                   |
 |                                                                          |
 | This program is distributed in the hope that it will be useful,          |
 | but WITHOUT ANY WARRANTY; without even the implied warranty of           |
 | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            |
 | GNU General Public License for more details.                             |
 |                                                                          |
 | You should have received a copy of the GNU General Public License        |
 | along with this program; if not, write to the Free Software              |
 | Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,USA.|
 +--------------------------------------------------------------------------+
*/

// pops up a window showing the file 'path', having 'title' as window title
function popUp(path,title,confirmation,question) {
    var popwin;
	var input=true;
	if (confirmation) input=confirm(question);
	if (input) {
	    if (title!="view") {
	        // small window
	        leftval=(screen.width)?(screen.width-400)/2:100;
	        topval=(screen.height)?(screen.height-500)/2:100;
	        popwin=window.open(path, title, "width=400,height=500,top=" + topval + ",left=" + leftval + ",toolbar=0,scrollbars=1,directories=no,location=0,statusbar=0,menubar=0,resizable=0");
	    } else {
	        // big window
	        leftval=(screen.width)?(screen.width-750)/2:100;
	        topval=(screen.height)?(screen.height-750)/2:100;
	        popwin=window.open(path, title, "width=750,height=750,top=" + topval + ",left=" + leftval + ",toolbar=0,scrollbars=1,directories=no,location=0,statusbar=0,menubar=0,resizable=1");
	    }
	    popwin.focus();
	}
}

// selects all entries in the selectbox 'select_name' in the form 'form_name'
function setSelect(form_name, select_name) {
    var select_field=document.forms[form_name].elements[select_name];
    for (var i=0; i < select_field.length; i++) select_field.options[i].selected=true;
    return true;
}

// opens a dialog box with 'title' as title
// when user clicks ok, he will be forwarded to file 'path'
function confirmClick(title,path) {
    var input=confirm(title);
    if (input) window.location.href=path;
}

// changes the color of table row 'tr'
function changeColor(row, new_color) {
    var cells=null;
    var i=null;

    // go sure that row is available
    if (typeof(row.style)=='undefined') return false;

    // trie to get all fields of current row
    if (typeof(document.getElementsByTagName)!='undefined') {
        cells=row.getElementsByTagName('td');
    } else if (typeof(row.cells)!='undefined') {
        cells=row.cells;
    } else {
        return false;
    }

    // change border color
    if (typeof(cells[0].style.setAttribute)!='undefined') {
    for (i=0; i<cells.length; i++) cells[i].style.setAttribute('borderColor',new_color,false);
    } else  if (typeof(cells[0].style.borderColor)!='undefined') {
        for (i=0; i<cells.length; i++) cells[i].style.borderColor=new_color;
    } else {
        return false;
    }
    return true;
} 

// changes the visibility attribute of an element to visible
function showElement(id)
{
  var element = document.getElementById(id);
    element.style.display = 'block';
}

// show ff add on internet explorers
window.onload = function() {	
	if(document.defaultCharset) showElement('ffadd');
}

</script>


src/phpMyBackupPro/global_conf.php
<?php

// This file is automatically generated and modified by phpMyBackupPro v.2.3

$CONF['sql_user']="";
$CONF['sql_passwd']="";
$CONF['sql_host']="localhost";
$CONF['sql_db']="";
$CONF['ftp_use']="0";
$CONF['allow_dir_backup']="0";
$CONF['email_use']="0";
$CONF['sitename']="mySite";
$CONF['lang']="english";
$CONF['import_error']="1";
$CONF['no_login']="0";
$CONF['login']="0";
$CONF['dir_backup']="0";
$CONF['dir_rec']="1";
$CONF['email']="";
$CONF['ftp_pasv']="1";
$CONF['ftp_del']="1";
$CONF['ftp_server']="example.net";
$CONF['ftp_user']="";
$CONF['ftp_passwd']="";
$CONF['ftp_path']="example/";
$CONF['ftp_port']="21";
$CONF['date']="%d.%m.%Y %H:%M Uhr";
$CONF['del_time']="7";
$CONF['del_number']="5";
$CONF['timelimit']="600";
$CONF['confirm']="1";
$CONF['stylesheet']="standard";

?>

src/phpMyBackupPro/sys_vars.inc.php
<?php
/*
 +--------------------------------------------------------------------------+
 | phpMyBackupPro                                                           |
 +--------------------------------------------------------------------------+
 | Copyright (c) 2004-2012 by Dirk Randhahn                                 |                               
 | http://www.phpMyBackupPro.net                                            |
 | version information can be found in definitions.php.                     |
 |                                                                          |
 | This program is free software; you can redistribute it and/or            |
 | modify it under the terms of the GNU General Public License              |
 | as published by the Free Software Foundation; either version 2           |
 | of the License, or (at your option) any later version.                   |
 |                                                                          |
 | This program is distributed in the hope that it will be useful,          |
 | but WITHOUT ANY WARRANTY; without even the implied warranty of           |
 | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            |
 | GNU General Public License for more details.                             |
 |                                                                          |
 | You should have received a copy of the GNU General Public License        |
 | along with this program; if not, write to the Free Software              |
 | Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,USA.|
 +--------------------------------------------------------------------------+
*/

// system variables are documented in documents/SYSTEM_VARIABLES.txt

// set general system variables 
$update=FALSE;
if (!isset($PMBP_SYS_VAR['last_scheduled'])) {
    $PMBP_SYS_VAR['last_scheduled']="";
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['this_login'])) {
    $PMBP_SYS_VAR['this_login']="";
    $update=TRUE;
}

if (!isset($PMBP_SYS_VAR['last_login'])) {
    $PMBP_SYS_VAR['last_login']="";
    $update=TRUE;
}
if (isset($CONF['sql_passwd']) && isset($CONF['sql_user'])) {
    if (!isset($PMBP_SYS_VAR['security_key']) && $CONF['sql_passwd'] && $CONF['sql_user']) {
        $PMBP_SYS_VAR['security_key']=md5($_SERVER['PHP_SELF'].$CONF['sql_passwd'].$CONF['sql_user']);
        $update=TRUE;
    }
}
if (!isset($PMBP_SYS_VAR['dir_lists'])) {
    $PMBP_SYS_VAR['dir_lists']=1;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['memory_limit'])) {
    $PMBP_SYS_VAR['memory_limit']=4000000; // (less than) 4 mb
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['except_tables'])) {
    $PMBP_SYS_VAR['except_tables']="";
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['scheduled_debug'])) {
    $PMBP_SYS_VAR['scheduled_debug']=0;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['schedule_all_dbs'])) {
    $PMBP_SYS_VAR['schedule_all_dbs']=0;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['ftp_timeout'])) {
    $PMBP_SYS_VAR['ftp_timeout']=10;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['dir_email_backup'])) {
    $PMBP_SYS_VAR['dir_email_backup']=0;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['C_disable_config'])) {
    $PMBP_SYS_VAR['C_disable_config']=0;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['C_disable_sqlquery'])) {
    $PMBP_SYS_VAR['C_disable_sqlquery']=0;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['B_backup_chmod'])) {
    $PMBP_SYS_VAR['B_backup_chmod']="0700";
    $update=TRUE;
}

// update login sys vars
if (isset($_GET['login']) || isset($_POST['login'])) {
	$login_str=strftime($CONF['date'],time())." (IP: ".$_SERVER['REMOTE_ADDR'].")";
	if ($login_str!=$PMBP_SYS_VAR['this_login'])
	{
	    $PMBP_SYS_VAR['last_login']=$PMBP_SYS_VAR['this_login'];
	    $PMBP_SYS_VAR['this_login']=$login_str;
	    $update=TRUE;
	}
}

// functions.inc.php
if (!isset($PMBP_SYS_VAR['F_dbs'])) {
    $PMBP_SYS_VAR['F_dbs']="";
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['F_comment'])) {
    $PMBP_SYS_VAR['F_comment']="";
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['F_tables'])) {
    $PMBP_SYS_VAR['F_tables']=1;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['F_data'])) {
    $PMBP_SYS_VAR['F_data']=1;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['F_drop'])) {
    $PMBP_SYS_VAR['F_drop']=1;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['F_compression'])) {
    $PMBP_SYS_VAR['F_compression']="";
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['F_ftp_dirs'])) {
    $PMBP_SYS_VAR['F_ftp_dirs']="";
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['F_ftp_dirs_2'])) {
    $PMBP_SYS_VAR['F_ftp_dirs_2']="";
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['F_packed'])) {
    $PMBP_SYS_VAR['F_packed']="";
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['F_updates'])) {
    $PMBP_SYS_VAR['F_updates']=1;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['F_ffadd'])) {
    $PMBP_SYS_VAR['F_ffadd']=1;
    $update=TRUE;
}

// scheduled.php
if (!isset($PMBP_SYS_VAR['EXS_scheduled_file'])) {
    $PMBP_SYS_VAR['EXS_scheduled_file']="???.php";
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['EXS_scheduled_dir'])) {
    $PMBP_SYS_VAR['EXS_scheduled_dir']=0;
    $update=TRUE;
}
if (!isset($PMBP_SYS_VAR['EXS_period'])) {
    $PMBP_SYS_VAR['EXS_period']="";
    $update=TRUE;
}

// global_conf.php
if (!isset($CONF['sitename'])) {
    $CONF['sitename']="mySite";
    $update=TRUE;
}
if (!isset($CONF['lang'])) {
    $CONF['lang']="english";
    $update=TRUE;
}
if (!isset($CONF['import_error'])) {
    $CONF['import_error']="1";
    $update=TRUE;
}
if (!isset($CONF['no_login'])) {
    $CONF['no_login']="0";
    $update=TRUE;
}
if (!isset($CONF['login'])) {
    $CONF['login']="0";
    $update=TRUE;
}
if (!isset($CONF['dir_backup'])) {
    $CONF['dir_backup']="0";
    $update=TRUE;
}
if (!isset($CONF['dir_rec'])) {
    $CONF['dir_rec']="1";
    $update=TRUE;
}
if (!isset($CONF['email_use'])) {
    $CONF['email_use']="0";
    $update=TRUE;
}
if (!isset($CONF['email'])) {
    $CONF['email']="";
    $update=TRUE;
}
if (!isset($CONF['ftp_use'])) {
    $CONF['ftp_use']="0";
    $update=TRUE;
}
if (!isset($CONF['ftp_pasv'])) {
    $CONF['ftp_pasv']="1";
    $update=TRUE;
}
if (!isset($CONF['ftp_del'])) {
    $CONF['ftp_del']="1";
    $update=TRUE;
}
if (!isset($CONF['ftp_server'])) {
    $CONF['ftp_server']="";
    $update=TRUE;
}
if (!isset($CONF['ftp_user'])) {
    $CONF['ftp_user']="";
    $update=TRUE;
}
if (!isset($CONF['ftp_passwd'])) {
    $CONF['ftp_passwd']="";
    $update=TRUE;
}
if (!isset($CONF['ftp_path'])) {
    $CONF['ftp_path']="";
    $update=TRUE;
}
if (!isset($CONF['ftp_port'])) {
    $CONF['ftp_port']="21";
    $update=TRUE;
}
if (!$_SESSION['multi_user_mode']) {
    if (!isset($CONF['sql_passwd'])) {
        $CONF['sql_passwd']="";
        $update=TRUE;
    }
    if (!isset($CONF['sql_host'])) {
        $CONF['sql_host']="localhost";
        $update=TRUE;
    }
    if (!isset($CONF['sql_user'])) {
        $CONF['sql_user']="";
        $update=TRUE;
    }
    if (!isset($CONF['sql_db'])) {
        $CONF['sql_db']="";
        $update=TRUE;
    }
}
if (!isset($CONF['date'])) {
    $CONF['date']="%x %X";
    $update=TRUE;
}
if (!isset($CONF['del_time'])) {
    $CONF['del_time']="7";
    $update=TRUE;
}
if (!isset($CONF['del_number'])) {
    $CONF['del_number']="5";
    $update=TRUE;
}
if (!isset($CONF['timelimit'])) {
    $CONF['timelimit']="60";
    $update=TRUE;
}
if (!isset($CONF['confirm'])) {
    $CONF['confirm']="1";
    $update=TRUE;
}
if (!isset($CONF['stylesheet'])) {
    $CONF['stylesheet']="standard";
    $update=TRUE;
}

// save global_conf.php
if ($update) PMBP_save_global_conf();

?>

src/phpMyBackupPro/login.php
<?php
/*
 +--------------------------------------------------------------------------+
 | phpMyBackupPro                                                           |
 +--------------------------------------------------------------------------+
 | Copyright (c) 2004-2012 by Dirk Randhahn                                 |                               
 | http://www.phpMyBackupPro.net                                            |
 | version information can be found in definitions.php.                     |
 |                                                                          |
 | This program is free software; you can redistribute it and/or            |
 | modify it under the terms of the GNU General Public License              |
 | as published by the Free Software Foundation; either version 2           |
 | of the License, or (at your option) any later version.                   |
 |                                                                          |
 | This program is distributed in the hope that it will be useful,          |
 | but WITHOUT ANY WARRANTY; without even the implied warranty of           |
 | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            |
 | GNU General Public License for more details.                             |
 |                                                                          |
 | You should have received a copy of the GNU General Public License        |
 | along with this program; if not, write to the Free Software              |
 | Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,USA.|
 +--------------------------------------------------------------------------+
*/

session_start();

// get password and username
require_once("definitions.php");

// login with http authentification
if ($CONF['login'] && !$_SESSION['multi_user_mode']) {
    if (!isset($_SERVER['PHP_AUTH_USER'])
    || (isset($_GET["login"]) && !($_SERVER['PHP_AUTH_USER']==$CONF['sql_user'] && $_SERVER['PHP_AUTH_PW']==$CONF['sql_passwd']))
    || (isset($_GET["logout"]) && $_SERVER['PHP_AUTH_PW']==$CONF['sql_passwd'])) {
        header("WWW-Authenticate: Basic realm=\"phpMyBackupPro\"");
        header("HTTP/1.0 401 Unauthorized");
        echo LI_NOT_LOGED_OUT;
        echo ": <a href=\"index.php?logout=TRUE\">".F_LOGOUT."</a><br>";
        echo LI_MSG.": <a href=\"index.php?login=TRUE\">".LI_LOGIN."</a>";
        exit;
    } else if ($_SERVER['PHP_AUTH_PW']!=$CONF['sql_passwd']) {
        echo LI_LOGED_OUT."<br>\n".LI_MSG;
        echo ": <a href=\"index.php?login=TRUE\">".LI_LOGIN."</a>";
        unset($_SESSION['LOGGED_IN']);
        unset($_SESSION['sql_host_org']);
        unset($_SESSION['sql_user_org']);
        unset($_SESSION['sql_passwd_org']);
        @session_destroy();
        exit;
    }
    
    // login with html authentification
} else {

    // disable login functions if $CONF['no_login'] is true
    if ($CONF['no_login']!="1") {
        if (!isset($_SESSION['LOGGED_IN'])) $_SESSION['LOGGED_IN']=FALSE;    

        // not logged in
        if (!$_SESSION['LOGGED_IN']) {
            if (!isset($_POST['password'])) $_POST['password']=FALSE;
            if (!isset($_POST['username'])) $_POST['username']=FALSE;            

            // distinguish between multi and single user mode
            if ($_SESSION['multi_user_mode'])
            {
                $con=@mysql_connect($PMBP_MU_CONF['sql_host_admin'],$PMBP_MU_CONF['sql_user_admin'],$PMBP_MU_CONF['sql_passwd_admin']);
                mysql_select_db("mysql");
                $res=mysql_query("select * from user where (User='".$_POST['username']."' or User='') and password=password('".$_POST['password']."')");
                $success=mysql_fetch_array($res);
                @mysql_close($con);
            } else {
                if ($CONF['sql_user']==$_POST['username'] && $CONF['sql_passwd']==$_POST['password']) {
                    $success=TRUE;
                } else {
                    $success=FALSE;
                }
            }
            // was the login successful
            if($success) {
                // login successful - set "LOGGED_IN"-bit
	            $_SESSION['LOGGED_IN']=TRUE;
	            $_GET['login']=TRUE;                

                // just make these settings for the first page after login, because definitions.php was already loaded
                $_SESSION['sql_user']=$_POST['username'];
                $_SESSION['sql_passwd']=$_POST['password'];
                $_SESSION['sql_host']=$PMBP_MU_CONF['sql_host_admin'];
                $CONF['sql_user']=$_SESSION['sql_user'];
                $CONF['sql_passwd']=$_SESSION['sql_passwd'];
                $CONF['sql_host']=$_SESSION['sql_host'];
                $CONF['sql_db']="";                
	        } else {
                // login failed - print login form
                ?>
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
   "http://www.w3.org/TR/html4/loose.dtd">
<html<?php echo ARABIC_HTML;?>>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=<?php echo BD_CHARSET_HTML;?>">
<link rel="stylesheet" href="<?php echo PMBP_STYLESHEET_DIR.$CONF['stylesheet'];?>.css" type="text/css">
<title>phpMyBackupPro <?php echo PMBP_VERSION; ?></title>
</head>
<body onLoad="document.login.username.focus()"><center>
<form name="login" action="" method="POST">
<table width="400">
  <tr>
    <th colspan="2" class="active">
<?php
echo PMBP_image_tag("logo.png","phpMyBackupPro","http://www.phpMyBackupPro.net","http://www.phpMyBackupPro.net");
?>
    </th>
  </tr>
  <tr>
    <td colspan="2"><?php echo LI_MSG; ?>:</td>
  </tr>
  <tr>
    <td><?php echo LI_USER; ?>:</td>
    <td><input type="text" name="username"></td>
  </tr>
  <tr>
    <td><?php echo LI_PASSWD; ?>:</td>
    <td><input type="password" name="password"></td>
  </tr>
  <tr>
    <td colspan="2"><input type="submit" name="login" value="<?php echo LI_LOGIN; ?>" class="button"></td>
  </tr>
</table>
</form>
</center>
</body>
</html>
	            <?php
	            $CONF="";
	            
	            // break loading page if not logged in
	            exit;
	        }
	    }
	
	    // logout
	    if (isset($_GET['logout'])) {
	        @session_start();
	        unset($_SESSION['PMBP_VERSION']);
	        unset($_SESSION['LOGGED_IN']);
	        unset($_SESSION['sql_host_org']);
	        unset($_SESSION['sql_user_org']);
	        unset($_SESSION['sql_passwd_org']);
	        unset($_SESSION['sql_user']);
	        unset($_SESSION['sql_passwd']);
            unset($_SESSION['file_system']);
	        header("Location: index.php");
	        exit;
	    }

    // when login functions is disabled
    } else {
        $_SESSION['LOGGED_IN']="Login deactivated!";
    }
} // end type of auth
?>
